app/models/job_template.rb

```diff
@@ -76,6 +79,39 @@ def self.import(template, options = {})
     template
   end
 
+  # Syncs a template with the given contents
+  def sync(template, options = {})
+    metadata = JobTemplate.parse_metadata(template)
+    return if metadata.blank? || metadata.delete('kind') != 'job_template'
+
+    new_input_hash = {}
+
+    metadata.delete('template_inputs').each do |input|
+      new_input_hash[input['name']] = input
+    end
+
+    self.assign_attributes(metadata.merge(:template => template.gsub(/<%\#.+?.-?%>\n?/m, '')).merge(options))
```

Comment:
Can we extract this and the part in `import` method into one place? Maybe there are more things that could be shared between `import` and `sync`. The import could be understand just as a sync with empty template.